soroban-sdk/src/prng.rs

```diff
@@ -0,0 +1,48 @@
+//! Prng contains functions for pseudo-random functions.
+use crate::{
+    env::internal, unwrap::UnwrapInfallible, Bytes, Env, IntoVal, TryIntoVal, Val,
+    Vec,
+};
+
+/// Prng provides access to pseudo-random  functions.
+pub struct Prng {
+    env: Env,
+}
+
+impl Prng {
+    pub(crate) fn new(env: &Env) -> Prng {
+        Prng { env: env.clone() }
+    }
+
+    pub fn env(&self) -> &Env {
+        &self.env
+    }
+
+    // Reseeds the pseudorandom number generator (PRNG) with the provided `seed` value.
+    pub fn prng_reseed(&self, seed: &Bytes) {
+        let env = self.env();
+        internal::Env::prng_reseed(env, seed.into()).unwrap_infallible();
+    }
+
+    // Returns a random u64 in the range between `lower` and `upper` inclusive.
+    pub fn prng_u64_in_inclusive_range(&self, lower: u64, upper: u64) -> u64 {
```

Comment:
@graydon Can we rename this function to `u64_in_range`, and replace the `lower: u64`, `upper:64` parameters with `r: impl RangeBounds<u64>`?
```suggestion
    pub fn u64_in_range(&self, r: impl RangeBounds<u64>) -> u64 {
```

It's allows developer to specify the random using rust range syntax, which supports both inclusive and exclusive ranges. Use use this pattern in the Vec type's slice function already. For an example for how it works see:
https://github.com/stellar/rs-soroban-sdk/blob/1def3c6e5e87acc71bd63c71295f9588edd3805b/soroban-sdk/src/vec.rs#L761-L774